cassandraprotocol/client/handshake.go

```diff
@@ -0,0 +1,69 @@
+package client
+
+import (
+	"fmt"
+	"github.com/datastax/go-cassandra-native-protocol/cassandraprotocol"
+	"github.com/datastax/go-cassandra-native-protocol/cassandraprotocol/frame"
+	"github.com/datastax/go-cassandra-native-protocol/cassandraprotocol/message"
+)
+
+func NewStartupRequest(c *CqlConnection, version cassandraprotocol.ProtocolVersion, streamId int16) *frame.Frame {
+	var startup *message.Startup
+	if c.codec.CompressionAlgorithm() == "NONE" {
+		startup = message.NewStartup()
+	} else {
+		startup = message.NewStartup(message.WithCompression(c.codec.CompressionAlgorithm()))
+	}
+	request, _ := frame.NewRequestFrame(version, streamId, false, nil, startup)
+	return request
+}
+
+func Handshake(client *CqlConnection, version cassandraprotocol.ProtocolVersion, streamId int16) error {
+	if err := client.Send(NewStartupRequest(client, version, streamId)); err != nil {
+		return err
+	} else if response, err := client.Receive(); err != nil {
+		return err
+	} else if _, ok := response.Body.Message.(*message.Ready); !ok {
+		return fmt.Errorf("expected READY, got %T", response.Body.Message)
+	} else {
+		return nil
+	}
+}
+
+func HandshakeAuth(client *CqlConnection, version cassandraprotocol.ProtocolVersion, streamId int16, username string, password string) error {
+	authenticator := &PlainTextAuthenticator{
+		username: username,
+		password: password,
+	}
+	if err := client.Send(NewStartupRequest(client, version, streamId)); err != nil {
+		return err
+	} else if response, err := client.Receive(); err != nil {
```

Comment:
I thought we would port most of the `TestClient` functionality to this repo but this client doesn't listen for responses in the background and map them according to the streamid so it's a bit different. Any reason why you didn't add that logic here?

Reply:
I didn't see a reason for doing it so far. Just a basic `Send` and a basic `Receive` method does pretty much everything I wanted to do and test. Handling stream ids sounds a bit like we are starting to create a true driver to me, which I didn't want to do for now. 

One thing to notice is that the `Receive` method blocks until it actually receives a frame. It wasn't problematic for me, but we might indeed have to wrap it in a goroutine if we need more complex test scenarios.

That said, I'm not at all against adding more features to this client, this is really just the minimum that allows us to talk to a server.

Reply:
I agree that it wouldn't make sense to do that right now but my impression was that we would just move the `TestClient` code here, not create a new client per say.

Reply:
I will just make the `TestClient` use this client internally then

Reply:
> we talked about moving the existing TestClient code here, not create a new one.

Indeed but a lot of that code was doing encoding and decoding so there was a lot of overlap. In the end I moved only the `Authenticator` part (thanks for that btw) and started the rest from scratch.